drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -67,13 +68,25 @@ static int stm32_temp_channel_get(const struct device *dev,
 		return -ENOTSUP;
 	}
 
+#ifdef CONFIG_SOC_SERIES_STM32L5X
+	BUILD_ASSERT(TEMPSENSOR_CAL_VREFANALOG != 0);
+	temp = ((float)data->raw * cfg->tsv_mv) / TEMPSENSOR_CAL_VREFANALOG;
+	temp -= *TEMPSENSOR_CAL1_ADDR;
+	temp *= (TEMPSENSOR_CAL2_TEMP - TEMPSENSOR_CAL1_TEMP);
+	temp /= (*TEMPSENSOR_CAL2_ADDR - *TEMPSENSOR_CAL1_ADDR);
+	temp += TEMPSENSOR_CAL1_TEMP; /* temperature offset: 30°C */
```

Comment:
I'm puzzled with this section using a different configuration scheme than other series.
I'd prefer to avoid that each new contributor  makes things in his own way

I understand that L5 is a different beast and has to be handled in a different way,
but I had a look and U5 will use the same IP.

So let's let's add a L5 dedicated compatible (that could be reused for U5 later and probably
other series). There are examples in other drivers (STM32 UART would be the most simple)
on how to handle various compatibles in a single driver.

Reply:
> I'm puzzled with this section using a different configuration scheme than other series. I'd prefer to avoid that each new contributor makes things in his own way

The existing implementation is specific to F1, F4 and F7 families and does not work on other series, like the F0, L4, L5 or WL. This PR is therefore definitely an improvement.

Reply:
Maybe I wasn't clear enough.
A dedicated algorithm is required, no discussion about that.
My point is this implementation is done using HAL defines, while initial one is based on device tree.

Reply:
I will do the changes after my holidays, i agree with you @erwango , just need some look into other drivers what you exactly, i can split the compatible, but I have only knowledge on l5, so can i make it l5 specific compatible? I anyhow need the includes from the hal since this algorithm works with factory calibrated values which i need from the device itself via the hal. Any suggestion would be welcome.

Reply:
> so can i make it l5 specific compatible

Sure, others will be able to use the same compatible.

> I anyhow need the includes from the hal since this algorithm works with factory calibrated values which i need from the device itself via the hal. 

Using device tree configuration method, you need to report values from HAL/ref man to device tree.

Reply:
@erwango, These are not constants for the series, but constants to the device specific. They contain pointer addresses to the calibrated values in production to the single device.  Therefore I cannot use the devicetree for this since you don't know them at compile time. I could add the memory addresses they point to to the devicetree, but I think that would be far from portable and would require too much knowledge of the user if ever needed to be changed

from stm32l5xx_ll_adc.h:
```
/* Temperature sensor */
#define TEMPSENSOR_CAL1_ADDR               ((uint16_t*) (0x0BFA05A8UL)) /* Internal temperature sensor, address of parameter TS_CAL1: On STM32L5, temperature sensor ADC raw data acquired at temperature  30 DegC (tolerance: +-5 DegC), Vref+ = 3.0 V (tolerance: +-10 mV). */
#define TEMPSENSOR_CAL2_ADDR               ((uint16_t*) (0x0BFA05CAUL)) /* Internal temperature sensor, address of parameter TS_CAL2: On STM32L5, temperature sensor ADC raw data acquired at temperature 110 DegC (tolerance: +-5 DegC), Vref+ = 3.0 V (tolerance: +-10 mV). */
#define TEMPSENSOR_CAL1_TEMP               (30L)                        /* Internal temperature sensor, temperature at which temperature sensor has been calibrated in production for data into TEMPSENSOR_CAL1_ADDR (tolerance: +-5 DegC) (unit: DegC). */
#define TEMPSENSOR_CAL2_TEMP               (110L)                       /* Internal temperature sensor, temperature at which temperature sensor has been calibrated in production for data into TEMPSENSOR_CAL2_ADDR (tolerance: +-5 DegC) (unit: DegC). */
#define TEMPSENSOR_CAL_VREFANALOG          (3000UL)                     /* Analog voltage reference (Vref+) voltage with which temperature sensor has been calibrated in production (+-10 mV) (unit: mV). */

```

Reply:
What we should store in device tree are not the calibrated values, but the pointer addresses (which are constant for each series, but may vary between series), and all the constants that you just listed.
This doesn't change anything to the algorithm you're using, just the place were you get the constants.

Btw, I had a look to reference manuals for a bunch of series and there are actually several series that could use the variant you're creating. Considering this, it would be more adapted to create a new compatible ("st,stm32-temp-v2" for instance) dedicated to this new variant. This new compatible could include `st,stm32-temp`, block the properties that are not common and add new required properties, cf https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/clock/st%2Cstm32wb-rcc.yaml for an example.
Then, you should use compatibles for configuration, rather than `CONFIG_SOC_SERIES_STM32L5X`.

Reply:
ok, i've split acually in a seperate compatible with the common parts split off. Let me know if you prefer another way. I think this approach makes it more readible, but if you don't agree, i can change it. Same applies for the compatible name.